packages/text-serializer/lib/serializeText.js

```diff
@@ -0,0 +1,41 @@
+function collectElementByIDs(element) {
+  const dataStructures = {};
+  const { parents } = element;
+
+  if (!parents || parents.isEmpty) {
+    return dataStructures;
+  }
+
+  const rootElement = parents.get(parents.length - 1);
+
+  if (rootElement) {
+    rootElement.recursiveChildren.forEach((element) => {
+      // eslint-disable-next-line no-underscore-dangle
+      const isNotEmptyStringElement = element._meta && element._meta.get('id');
+
+      if (isNotEmptyStringElement) {
+        dataStructures[element.id.toValue()] = element;
+      }
+    });
+  }
+
+  return dataStructures;
+}
+
+const isPrimitive = e => (e !== Object(e));
```

Comment:
I think there might be one case where this won't work as intended, and that will be "undefined". Although I am not 100% sure if it is possible, but I think `valueOf` might return undefined in some cases where there isn't a value. Perhaps an enum element which doesn't contain any enumerations (`new namespace.elements.Enum()`) thus there isn't any possibly valid value to return.

Perhaps it would be good to have a test case for an empty enum to verify this case.

Reply:
Addressed on 2a32fc3